server/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -301,6 +320,9 @@ private void sendResponse(
         final TimeoutState timeoutState,
         final ActionListener<ClusterHealthResponse> listener
     ) {
+        // Creating the ClusterHealthResponse below can be computationally heavy. Ensure this thread is not running on a hot path thread.
```

Comment:
> Ensure this thread is not running on a hot path thread.

nit "Hot path thread" sounds fuzzy. At least I am not sure what to understand. Maybe removing this sentence is fine, as the rest of the comment and the line below are enough.